client/my-sites/domains/domain-management/settings/cards/registered-domain-details.tsx

```diff
@@ -72,11 +72,11 @@ const RegisteredDomainDetails = ( {
 				getRenewalPrice( purchase ) +
 				( domain.isRedeemable && redemptionProduct ? redemptionProduct.cost : 0 );
 			const currencyCode = purchase.currencyCode;
-			formattedPrice = formatCurrency( renewalPrice, currencyCode, { stripZeros: true } )!;
+			formattedPrice = formatCurrency( renewalPrice, currencyCode, { stripZeros: true } ) ?? '';
```

Comment:
no-null assertions are banned by our linting rules, and they are dangerous anyway. By adding a fallback we provide the same silent failure without needing to override TypeScript.